packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -0,0 +1,166 @@
+
+    /// @notice Transient storage slot that the timestamp for an Identifier is stored at.
+    ///         Equal to bytes32(uint256(keccak256("crossl2inbox.identifier.timestamp")) - 1)
+    bytes32 internal constant TIMESTAMP_SLOT = 0x2e148a404a50bb94820b576997fd6450117132387be615e460fa8c5e11777e02;
+
+    /// @notice Transient storage slot that the chainId for an Identifier is stored at.
+    ///         Equal to bytes32(uint256(keccak256("crossl2inbox.identifier.chainid")) - 1)
+    bytes32 internal constant CHAINID_SLOT = 0x6e0446e8b5098b8c8193f964f1b567ec3a2bdaeba33d36acb85c1f1d3f92d313;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Enforces that cross domain message sender and source are set. Reverts if not.
+    ///         Used to differentiate between 0 and nil in transient storage.
+    modifier notEntered() {
+        if (TransientContext.get(ENTERED_SLOT) == 0) revert NotEntered();
+        _;
+    }
+
+    /// @notice Returns the origin address of the Identifier. If not entered, reverts.
+    /// @return Origin address of the Identifier.
+    function origin() external view notEntered returns (address) {
+        return address(uint160(TransientContext.get(ORIGIN_SLOT)));
+    }
+
+    /// @notice Returns the block number of the Identifier. If not entered, reverts.
+    /// @return Block number of the Identifier.
+    function blockNumber() external view notEntered returns (uint256) {
+        return TransientContext.get(BLOCK_NUMBER_SLOT);
+    }
+
+    /// @notice Returns the log index of the Identifier. If not entered, reverts.
+    /// @return Log index of the Identifier.
+    function logIndex() external view notEntered returns (uint256) {
+        return TransientContext.get(LOG_INDEX_SLOT);
+    }
+
+    /// @notice Returns the timestamp of the Identifier. If not entered, reverts.
+    /// @return Timestamp of the Identifier.
+    function timestamp() external view notEntered returns (uint256) {
+        return TransientContext.get(TIMESTAMP_SLOT);
+    }
+
+    /// @notice Returns the chain ID of the Identifier. If not entered, reverts.
+    /// @return _chainId The chain ID of the Identifier.
+    function chainId() external view notEntered returns (uint256) {
+        return TransientContext.get(CHAINID_SLOT);
+    }
+
+    /// @notice Executes a cross chain message on the destination chain.
+    /// @param _id      Identifier of the message.
+    /// @param _target  Target address to call.
+    /// @param _message Message payload to call target with.
+    function executeMessage(
+        Identifier calldata _id,
+        address _target,
+        bytes memory _message
+    )
+        external
+        payable
+        reentrantAware
+    {
+        if (_id.timestamp > block.timestamp) revert InvalidIdTimestamp(_id.timestamp, block.timestamp);
```

Comment:
Its not clear to me why you would use a custom revert with arguments or not. It seems only useful in the case of a caller needing to get information and act on it